hal/src/rtl872x/ble_hal.cpp

```diff
@@ -1474,23 +2093,41 @@ void BleGap::handleConnectionStateChanged(uint8_t connHandle, T_GAP_CONN_STATE n
             le_get_conn_param(GAP_PARAM_CONN_INTERVAL, &connParams.max_conn_interval, connHandle);
             le_get_conn_param(GAP_PARAM_CONN_LATENCY, &connParams.slave_latency, connHandle);
             le_get_conn_param(GAP_PARAM_CONN_TIMEOUT, &connParams.conn_sup_timeout, connHandle);
-            LOG_DEBUG(TRACE, "[BLE peripheral] Connected, interval:0x%x, latency:0x%x, timeout:0x%x",
+            LOG_DEBUG(TRACE, "Connected, interval:0x%x, latency:0x%x, timeout:0x%x",
                             connParams.max_conn_interval, connParams.slave_latency, connParams.conn_sup_timeout);
-            connInfo_.version = BLE_API_VERSION;
-            connInfo_.size = sizeof(hal_ble_conn_info_t);
-            connInfo_.role = BLE_ROLE_PERIPHERAL;
-            connInfo_.conn_handle = connHandle;
-            connInfo_.conn_params = connParams;
-            connInfo_.address = peerAddr;
-            connInfo_.att_mtu = BLE_DEFAULT_ATT_MTU_SIZE; // Use the default ATT_MTU on connected.
-            for (const auto& callback : periphEvtCallbacks_) {
-                if (callback.first) {
-                    hal_ble_link_evt_t evt = {};
-                    evt.type = BLE_EVT_CONNECTED;
-                    evt.conn_handle = connInfo_.conn_handle;
-                    evt.params.connected.info = &connInfo_;
-                    callback.first(&evt, callback.second);
-                }
+
+            BleConnection connection = {};
+            connection.info.version = BLE_API_VERSION;
+            connection.info.size = sizeof(hal_ble_conn_info_t);
+            if (addressEqual(connectingAddr_, peerAddr)) {
```

Comment:
This check seems to be failing for me. In central role I'm getting `Connected as Peripheral` in the log and device crashes afterwards.